processor/processor.go

```diff
@@ -558,10 +559,13 @@ func (wp *workerPool) download(ctx context.Context, j *job.Job, validator *mimet
 		return derrors.E("creating request", err)
 	}
 
-	if j.UserAgent != "" {
-		req.Header.Set("User-Agent", j.UserAgent)
-	} else if wp.p.UserAgent != "" {
-		req.Header.Set("User-Agent", wp.p.UserAgent)
+	// Merge job's request headers to processor
+	for k, v := range j.RequestHeaders {
+		wp.p.RequestHeaders[k] = v
```

Comment:
I believe this is wrong. `wp.p.RequestHeaders` contains the default headers for *every* job. By mutating it here, you're causing the headers of a *specific* job to get applied to *all* subsequent jobs.

Reply:
Ah yes! You are right! Sorry about that! Fixed!